### ISSUE TITLE
Deployment issue on artifactory

### ISSUE THREAD

**Issue by JGroselle** [jfrog/bamboo-artifactory-plugin#18]
Hello everyone,

We are trying to upgrade bamboo but we are facing an artifactory plugin issue.
The plugin doesn't deploy release in the selected artifactory repo but keep using the one defined by default in the plan job instead. Note that we use maven.
Bamboo version: 5.9.4 and 5.9.7
Plugin version: 1.8.0, 1.8.1, 1.8.2 and 1.9.0

Any help or comment appreciated.
Thank you.

**Comment by eyalbe4** [comment]
@JGroselle, we are looking into this and will upstate you soon. Thanks for reporting this.

**Comment by 33man** [comment]
Hi, 

just adding some info, the buildinfo while performing the release set correctly the bamboo variable to the selected repo in the Artifactory Maven3 Task.

Ex: bamboo_releaseRepoKey=aRepoOnlyAcceptingReleasedArtifact-releases-local

But the in the task we have by default snapshot repo :

![image](https://cloud.githubusercontent.com/assets/7490114/11588339/ed41c974-9a7f-11e5-8d50-068c66b4ec15.png)

And in the release page we select upon the release the release repo :
![image](https://cloud.githubusercontent.com/assets/7490114/11588370/2da8fe88-9a80-11e5-9208-47d001990a0b.png)

But when the task is launched the default "lib-snapshot-local" is taken instead of the one selected for performing the release "lib-releases-local".

I hope this will help to sort the issue.

**Comment by eyalbe4** [comment]
@JGroselle and @33man, 
I was unable to replicate this issue. I used version 1.9.0 of the plugin with Bamboo 5.9.2:
1. I create a maven and a gradle task, each in a separate plan.
2. I configured each of the tasks to publish to a snapshot repository in Artifactory.
3. After a successful build, in the "Artifactory Pro Release Staging" I selected a release Artifactory repositories and had my release artifacts published to the selected release repository.
Is there something I missed in your replication steps?
Did the "Artifactory Pro Release Staging" page render properly? Do you see any errors in the Bamboo logs? Do you see the "Release Value" and "Next Integration Value" fields with the suggested release and snapshot versions?
Please share what you see - we want to help you perform the upgrade.

**Comment by eyalbe4** [comment]
OK - I managed to reproduce the issue now. Will update soon.

**Comment by JGroselle** [comment]
Ok, quite happy to read that you have pointed the problem too.
We are waiting on you, you will be our savior ;)
Thank you for your time.

**Comment by 33man** [comment]
@eyalbe4 Hi, thanks for the support, this is what we have upon a successful build. (The default target repo (Publishing Repository) is a snapshot repo (libs-snapshots-local) but upon a release we select manually the releases only repo (libs-releases-local) but this change was never taken in account and artifactory return a HTTP 409 error as the plugin try to deploy a Release dependency into a snapshot only repository)

![image](https://cloud.githubusercontent.com/assets/7490114/11632444/7827dcca-9d07-11e5-8595-2ee71675f6d3.png)

**Comment by eyalbe4** [comment]
I created a snapshot version on top of release 1.9.0 of the plugin with a fix for this issue.
I added the download link to this version as a comment on [BAP-312](https://www.jfrog.com/jira/browse/BAP-312)
@JGroselle and @33man, please try it out and let me know if it fixes the issue for you.

**Comment by 33man** [comment]
Hi,

just tested the default task with snapshot repository as default for compilation and then i triggered the  release selecting a different repository and now it works fine !

I only see that some keys have changed in the buildinfo file 

1.9.0
"buildInfo.env.bamboo_releaseRepoKey" : "libs-releases-local"
"buildInfo.env.releaseRepoKey" : "libs-releases-local"

1.9.X
"buildInfo.env.bamboo_release_management_repoKey" : "libs-releases-local"
"buildInfo.env.release.management.repoKey" : "libs-releases-local"


Thanks for your quick help !

**Comment by JGroselle** [comment]
Hello @eyalbe4, for what we have tested now it works !
A huge thank you for your time !

**Comment by eyalbe4** [comment]
Anytime guys :)
